src/order/monotone/monovary.lean

```diff
@@ -271,3 +311,80 @@ protected lemma antivary_on_comm : antivary_on f g s ↔ antivary_on g f s :=
 ⟨antivary_on.symm, antivary_on.symm⟩
 
 end linear_order
+
+section
+variables [linear_order α] [linear_order β] (f : ι → α) (g : ι → β) {s : set ι}
+
+/-- If `f : ι → α` and `g : ι → β` are monovarying, then `monovary_order f g` is a linear order on
+`ι` that makes `f` and `g` simultaneously monotone. -/
+def monovary_order (i j : ι) : Prop :=
+prod.lex (<) (prod.lex (<) well_ordering_rel) (f i, g i, i) (f j, g j, j)
+
+instance : is_strict_total_order ι (monovary_order f g) :=
+{ trichotomous := λ i j, begin
+    convert trichotomous_of (prod.lex (<) $ prod.lex (<) well_ordering_rel) _ _,
+    { simp only [prod.ext_iff, ←and_assoc, imp_and_distrib, eq_iff_iff, iff_and_self],
+      exact ⟨congr_arg _, congr_arg _⟩ },
```

Comment:
I would expect this to be easier, since intuitively `prod.lex r s` should break any "ties" where `r` is not trichotomous is but `s` is. In other words, wouldn't we be able to strengthen [`prod.is_trichotomous`](https://leanprover-community.github.io/mathlib_docs/data/prod/basic.html#prod.is_trichotomous) to only assume `is_irrefl r`?

In particular, because we do not assume `f` and `g` monovary here, I don't see any special conditions on the order induced by `f` or `g`.

Reply:
If `r` isn't trichotomous, say `a` and `b` are incomparable, then `(a, c)` and `(b, c)` are incomparable (`a` isn't equal to `b`, we don't have `r a b` nor `r b a`). Does that answer it all?

Reply:
I see, you're right. So the reason this works is that, while `f` and `g` need not be injective, they are not partial.